api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/MISAUKeyPopsCohortQueries.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License Version
+ * 1.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for
+ * the specific language governing rights and limitations under the License.
+ *
+ * Copyright (C) OpenMRS, LLC. All Rights Reserved.
+ */
+package org.openmrs.module.eptsreports.reporting.library.cohorts;
+
+import java.util.Date;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.CompositionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/** Defines all of the TxCurrCohortQueries we want to expose for EPTS */
+@Component
+public class MISAUKeyPopsCohortQueries {
+
+  @Autowired private ResumoMensalCohortQueries resumoMensalCohortQueries;
```

Comment:
Now this being a new class, can we adopt the bean creation through a constructor with only one @Autowired place at the method signature? this the recommended srping way, reduces on time for initial start up and can be called directitly in the tests classes

Reply:
Thanks Nicholas, please check now.